lib/assets/uploader.js

```diff
@@ -0,0 +1,176 @@
+
+function setApplicationName(name, version) {
+  window.apiVideoFlutterUploader.params.application = {
+    name,
+    version
+  }
+}
+
+function setChunkSize(chunkSize) {
+  window.apiVideoFlutterUploader.params.chunkSize = chunkSize;
+}
+
+function getProgressiveSession(sessionId) {
+  return window.apiVideoFlutterUploader.progressiveSessions[sessionId];
+}
+
+function setProgressiveSession(sessionId, progressiveSession) {
+  window.apiVideoFlutterUploader.progressiveSessions = window.apiVideoFlutterUploader.progressiveSessions || {};
+  window.apiVideoFlutterUploader.progressiveSessions[sessionId] = progressiveSession;
+}
+
+function storeStandardUploader(uploader) {
+  window.apiVideoFlutterUploader.standardUploaders = window.apiVideoFlutterUploader.standardUploaders || [];
+  window.apiVideoFlutterUploader.standardUploaders.push(uploader);
+}
+
+function createProgressiveUploadWithUploadTokenSession(sessionId, uploadToken, videoId) {
+  return progressiveUploadCreationHelper({
+    sessionId,
+    uploadToken,
+    videoId: videoId || undefined,
+  });
+}
+
+function createProgressiveUploadWithApiKeySession(sessionId, apiKey, videoId) {
+  return progressiveUploadCreationHelper({
+    sessionId,
+    apiKey,
+    videoId: videoId || undefined,
+  });
+}
+
+async function uploadPart(sessionId, filePath, onProgress) {
+  return await uploadPartHelper(sessionId, filePath, onProgress, async (session, blob) => {
+    return await session.uploader.uploadPart(blob);
+  });
+}
+
+async function uploadLastPart(sessionId, filePath, onProgress) {
+  return await uploadPartHelper(sessionId, filePath, onProgress, async (session, blob) => {
+    return await session.uploader.uploadLastPart(blob);
+  });
+}
+
+async function getBlobFromPath(filePath) {
+  return await fetch(filePath)
+    .then(r => r.blob());
+}
+
+async function cancelAll() {
+  const sessions = window.apiVideoFlutterUploader.progressiveSessions;
+  if (sessions != null) {
+    Object.values(sessions).forEach(session => {
+      session.uploader.cancel();
+    });
+  }
+
+  const standardUploaders = window.apiVideoFlutterUploader.standardUploaders;
+  if (standardUploaders != null) {
+    standardUploaders.forEach(uploader => {
+      uploader.cancel();
+    });
+  }
+}
+
+async function jsDisposeProgressiveUploadSession(sessionId) {
```

Comment:
This method should be called `disposeProgressiveUploadSession` instead

Reply:
:+1: well observed